migrations/versions/1060_add_external_id_column.py

```diff
@@ -0,0 +1,77 @@
+"""Add external ID column
+
+Revision ID: 1050
+Revises: 1040
+Create Date: 2017-11-23 16:46:00.569239
+
+"""
+from alembic import op
+import json
+import random
+import sqlalchemy as sa
+from sqlalchemy.sql import text
+
+# revision identifiers, used by Alembic.
+revision = '1060'
+down_revision = '1050'
+
+DIRECT_AWARD_AUDIT_TYPES = ('create_project', 'create_project_search', 'lock_project', 'downloaded_project')
+
+
+def random_positive_external_id():
+    return random.SystemRandom().randint(10 ** 14, (10 ** 15) - 1)
+
+
+def upgrade():
+    op.add_column('direct_award_projects', sa.Column('external_id', sa.BIGINT()))
+    op.create_index(op.f('ix_direct_award_projects_external_id'), 'direct_award_projects', ['external_id'], unique=True)
+
+    # Manually generate random IDs for all existing saved searches
+    project_connection = op.get_bind()
+    project_id_internal_to_external = {}
+    for row_id, *_ in project_connection.execute('select id from direct_award_projects'):
+        # Some logic to ensure all IDs generated are unique; keep track so we can update AuditEvents.
+        random_id = random_positive_external_id()
+        while random_id in project_id_internal_to_external:
```

Comment:
This still won't work with a dict - should be `project_id_internal_to_external.values()`.